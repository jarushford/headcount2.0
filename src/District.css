.district-card {
  border: 2px solid #333;
  border-radius: 3px;
  box-shadow: 1px 1px 3px 1px rgba(0, 0, 0, 0.25);
  padding: 4px;
}

hr {
  width: 75%;
}

h2 {
  font-size: 20px;
  text-align: center;
}

li {
  list-style: none;
}