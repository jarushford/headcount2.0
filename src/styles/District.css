.district-card {
  box-sizing: border-box;
  border-radius: 3px;
  box-shadow: 1px 1px 5px 1px rgba(0, 0, 0, 0.3);
  padding: 4px;
  background: url(http://www.fbcgallatin.org/wp-content/uploads/2016/10/chalkboard.jpg);
  background-position: center;
  background-size: cover;
  background-color: rgba(0, 0, 0, 0.4);
  background-blend-mode: darken;
  color: white;
  width: 340px;
  margin: 0 auto;
  cursor: pointer;
}

hr {
  width: 85%;
}

h2 {
  font-family: 'Chalk';
  font-size: 16px;
  letter-spacing: 2px;
  text-align: left;
  margin-left: 8%;
}

ul {
  padding-left: 8%;
}

li {
  font-family: monospace;
  list-style: none;
  font-size: 1.4rem;
  width: 100%;
  display: flex;
  justify-content: flex-start;
  align-items: center;
}

svg {
  margin: 8px 4px -3px 10px;
}

.line-accent {
  stroke-width: 40px;
  stroke-dasharray: 150%;
  stroke: rgb(0, 97, 0);
  stroke-dashoffset: 0;
}

.red-accent {
  stroke: rgb(129, 0, 0);
}

.line-bg {
  stroke-width: 40px;
  stroke: #D9E4EA;
}

.data-span {
  font-size: .8rem;
}

.selected {
  border: 4px solid rgb(236, 217, 42);
}
