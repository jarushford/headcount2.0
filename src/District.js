import React from 'react'
import { uid } from 'react-uid'
import PropTypes from 'prop-types'
import './styles/District.css'

export default function District(props) {
  const { districtData, compareDistrict, selected } = props
  const { location, stats } = districtData
  const statsKeys = Object.keys(stats)
  const statsArray = statsKeys.map((year) => {
    return (
      <li key={uid(year)}>
        {year}
        <svg height="30" width="150">
          <line className="line-bg" x1={stats[year] * 145} y1="0" x2="145" y2="0" />
          <line className={`line-accent ${stats[year] < 0.5 && 'red-accent'}`} x1="0" y1="0" x2={stats[year] * 145} y2="0" />
        </svg>
        <span className="data-span">{stats[year]}</span>
      </li>
    )
  })

  return (
    <div
      className={`district-card ${selected && 'selected'}`}
      onClick={() => {
        compareDistrict(location)
      }}
    >
      <h2>{location}</h2>
      <hr />
      <ul>{statsArray}</ul>
    </div>
  )
}


District.propTypes = {
  districtData: PropTypes.shape({ location: '', stats: PropTypes.shape({ 2005: 0 }) }).isRequired,
  compareDistrict: PropTypes.func.isRequired,
  selected: PropTypes.bool.isRequired
}
