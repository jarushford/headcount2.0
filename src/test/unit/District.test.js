import React from 'react';
import District from '../../District';
import { shallow } from 'enzyme'

describe('App', () => {
  const mockCompare = jest.fn()
  const data = { location: 'Colorado', stats: { 2006: 0 } }
  const wrapper = shallow(<District districtData={data}
    compareDistrict={mockCompare}
    selected={false}/>)

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot when selected', () => {
    const wrapper2 = shallow(<District districtData={data}
    compareDistrict={mockCompare}
    selected={true}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should compare a card when clicked on', () => {
    wrapper.find('.district-card').simulate('click')

    expect(mockCompare).toBeCalled()
  })

})