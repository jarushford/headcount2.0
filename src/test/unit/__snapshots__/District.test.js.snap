// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <District
    districtData={
        Object {
            "location": "Colorado",
            "stats": Object {
              "2006": 0,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Colorado
</h2>,
        <hr />,
        <ul>
          <li>
                    2006
                    <svg
                              height="30"
                              width="150"
                    >
                              <line
                                        className="line-bg"
                                        x1={0}
                                        x2="145"
                                        y1="0"
                                        y2="0"
                              />
                              <line
                                        className="line-accent red-accent"
                                        x1="0"
                                        x2={0}
                                        y1="0"
                                        y2="0"
                              />
                    </svg>
                    <span
                              className="data-span"
                    />
          </li>
</ul>,
      ],
      "className": "district-card undefined",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Colorado",
        },
        "ref": null,
        "rendered": "Colorado",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              2006
              <svg
                            height="30"
                            width="150"
              >
                            <line
                                          className="line-bg"
                                          x1={0}
                                          x2="145"
                                          y1="0"
                                          y2="0"
                            />
                            <line
                                          className="line-accent red-accent"
                                          x1="0"
                                          x2={0}
                                          y1="0"
                                          y2="0"
                            />
              </svg>
              <span
                            className="data-span"
              />
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "val-2006",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2006",
                <svg
                  height="30"
                  width="150"
>
                  <line
                                    className="line-bg"
                                    x1={0}
                                    x2="145"
                                    y1="0"
                                    y2="0"
                  />
                  <line
                                    className="line-accent red-accent"
                                    x1="0"
                                    x2={0}
                                    y1="0"
                                    y2="0"
                  />
</svg>,
                <span
                  className="data-span"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "2006",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <line
                      className="line-bg"
                      x1={0}
                      x2="145"
                      y1="0"
                      y2="0"
/>,
                    <line
                      className="line-accent red-accent"
                      x1="0"
                      x2={0}
                      y1="0"
                      y2="0"
/>,
                  ],
                  "height": "30",
                  "width": "150",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "line-bg",
                      "x1": 0,
                      "x2": "145",
                      "y1": "0",
                      "y2": "0",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "line",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "line-accent red-accent",
                      "x1": "0",
                      "x2": 0,
                      "y1": "0",
                      "y2": "0",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "line",
                  },
                ],
                "type": "svg",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                  "className": "data-span",
                },
                "ref": null,
                "rendered": 0,
                "type": "span",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Colorado
</h2>,
          <hr />,
          <ul>
            <li>
                        2006
                        <svg
                                    height="30"
                                    width="150"
                        >
                                    <line
                                                className="line-bg"
                                                x1={0}
                                                x2="145"
                                                y1="0"
                                                y2="0"
                                    />
                                    <line
                                                className="line-accent red-accent"
                                                x1="0"
                                                x2={0}
                                                y1="0"
                                                y2="0"
                                    />
                        </svg>
                        <span
                                    className="data-span"
                        />
            </li>
</ul>,
        ],
        "className": "district-card undefined",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Colorado",
          },
          "ref": null,
          "rendered": "Colorado",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                2006
                <svg
                                height="30"
                                width="150"
                >
                                <line
                                                className="line-bg"
                                                x1={0}
                                                x2="145"
                                                y1="0"
                                                y2="0"
                                />
                                <line
                                                className="line-accent red-accent"
                                                x1="0"
                                                x2={0}
                                                y1="0"
                                                y2="0"
                                />
                </svg>
                <span
                                className="data-span"
                />
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "val-2006",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2006",
                  <svg
                    height="30"
                    width="150"
>
                    <line
                                        className="line-bg"
                                        x1={0}
                                        x2="145"
                                        y1="0"
                                        y2="0"
                    />
                    <line
                                        className="line-accent red-accent"
                                        x1="0"
                                        x2={0}
                                        y1="0"
                                        y2="0"
                    />
</svg>,
                  <span
                    className="data-span"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "2006",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <line
                        className="line-bg"
                        x1={0}
                        x2="145"
                        y1="0"
                        y2="0"
/>,
                      <line
                        className="line-accent red-accent"
                        x1="0"
                        x2={0}
                        y1="0"
                        y2="0"
/>,
                    ],
                    "height": "30",
                    "width": "150",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "line-bg",
                        "x1": 0,
                        "x2": "145",
                        "y1": "0",
                        "y2": "0",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "line",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "line-accent red-accent",
                        "x1": "0",
                        "x2": 0,
                        "y1": "0",
                        "y2": "0",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "line",
                    },
                  ],
                  "type": "svg",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0,
                    "className": "data-span",
                  },
                  "ref": null,
                  "rendered": 0,
                  "type": "span",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
